docs/using-injected-bindings/using-injected-bindings.md

```diff
@@ -0,0 +1,165 @@
+If both are set then the
+`SERVICE_BINDING_ROOT` environment variable takes the higher precedence.
+
+The following table summarizes how the final bind path is computed:
+
+| .spec.mountPath | SERVICE_BINDING_ROOT  | Final Bind Path                      |
+| --------------- | --------------------- | ------------------------------------ |
+| nil             | non-existent          | /bindings/ServiceBinding_Name        |
+| nil             | /some/path/root       | /some/path/root/ServiceBinding_Name  |
+| /home/foo       | non-existent          | /home/foo                            |
+| /home/foo       | /some/path/root       | /some/path/root/ServiceBinding_Name  |
+
+You can use the built-in language feature of your programming language of choice
+to read environment variables.
+
+For accessing bindings within `SERVICE_BINDING_ROOT`, there are language or
+framework specific programs:
+
+1. Python: https://github.com/baijum/pyservicebinding
+2. Go: https://github.com/baijum/servicebinding
+3. Java/Spring: https://github.com/spring-cloud/spring-cloud-bindings
+4. Quarkus: https://quarkus.io/guides/deploying-to-kubernetes#service-binding
+5. JS/Node: https://github.com/nodeshift/kube-service-bindings
+
+Note: All these libraries expect `SERVICE_BINDING_ROOT` to be set.
+
+Here is example Python client usage:
+
+```
+from pyservicebinding import binding
+try:
+    sb = binding.ServiceBinding()
+except binding.ServiceBindingRootMissingError as msg:
+    # log the error message and retry/exit
+    print("SERVICE_BINDING_ROOT env var not set")
+
+sb = binding.ServiceBinding()
+
+bindings_list = sb.bindings("postgresql")
+```
+
+In the above example, the `bindings_list` contains bindings for `postgresql`
+type.  For full API, see the
+[documentation](https://github.com/baijum/pyservicebinding).
+
+
+### Programming language specific library APIs
+
+The application can retrieve the bindings through the library available for your
+programming language of choice.  The language-specific APIs are encouraged to
+follow the pattern described here.  This is not an exhaustive list of APIs.
+Consult your library API documentation to confirm the behavior.
+
+The language binding API can create separate functions to retrieve bindings. For
+example, in Go:
+
+```
+Bindings(_type string) []map[string]string
+BindingsWithProvider(_type, provider string) []map[string]string
+```
+
+Languages with function overloading can use a method with different parameters
+to retrieve bindings. For example, in Java:
+
+```
+public List<Binding> filterBindings(@Nullable String type)
+public List<Binding> filterBindings(@Nullable String type, @Nullable String provider)
+```
+
+(Example taken from [Spring Cloud
+Bindings](https://github.com/spring-cloud/spring-cloud-bindings))
+
+Since the APIs are returning more than one value, depending on your application
+need, you can choose to connect to the first entry or all of them, if required.
+
+### Environment Variables
```

Comment:
I think giving an example would be nice here.

Reply:
Question: 

-  This content "The container must restart to update the values of environment variables if there is a change in the Secret resource." -- Is it relevant and applicable to Red Hat SBO? If yes, then move _only this sentence_ to the most suitable place where we talk about environment variables as Note. Suggest removing the rest of the section as it is redundant and we have explained it already.

Suggestion:

-  If the sentence discussed above does not apply to Red Hat SBO,  suggest removing this entire section as it is redundant and we have explained it already.

Reply:
The idea is to show some example usage.  The user still needs to figure out if for their respective languages.